core/txpool/legacypool/heap.go

```diff
@@ -0,0 +1,171 @@
+package legacypool
+
+import (
+	"container/heap"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// txHeapItem implements the Interface interface (https://pkg.go.dev/container/heap#Interface) of heap so that it can be heapified
+type txHeapItem struct {
+	tx        *types.Transaction
+	timestamp int64 // Unix timestamp (nanoseconds) of when the transaction was added
+	index     int
+}
+
+type txHeap []*txHeapItem
+
+func (h txHeap) Len() int { return len(h) }
+func (h txHeap) Less(i, j int) bool {
+	return h[i].timestamp < h[j].timestamp
+}
+func (h txHeap) Swap(i, j int) {
+	if i < 0 || j < 0 || i >= len(h) || j >= len(h) {
+		return // Silently fail if indices are out of bounds
+	}
+	h[i], h[j] = h[j], h[i]
+	if h[i] != nil {
+		h[i].index = i
+	}
+	if h[j] != nil {
+		h[j].index = j
+	}
+}
+
+func (h *txHeap) Push(x interface{}) {
+	item, ok := x.(*txHeapItem)
+	if !ok {
+		return
+	}
+	n := len(*h)
+	item.index = n
+	*h = append(*h, item)
+}
+
+func (h *txHeap) Pop() interface{} {
+	old := *h
+	n := len(old)
+	if n == 0 {
+		return nil // Return nil if the heap is empty
+	}
+	item := old[n-1]
+	old[n-1] = nil // avoid memory leak
+	*h = old[0 : n-1]
+	if item != nil {
+		item.index = -1 // for safety
+	}
+	return item
+}
+
+type TxOverflowPoolHeap struct {
```

Comment:
TxOverflowPoolHeap is not a heap at all, it's a pool
I suggest:
1. remove file overflowpool.go
2. rename TxOverflowPoolHeap --> TxOverflowPool
     ranme heap.go --> tx_overflowpool.go, ranme heap_test.go --> tx_overflowpool_test.go
3.  define localBufferPool as TxOverflowPool directly